### PR TITLE
Jetpack Section: Update My Site Menu Items Strings

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #13629
Related #13969

This PR updates the following for the `My Site` screen menu items:
- The `Activity log` string to `Activity Log` (title case).
- The `Jetpack settings` string to `Jetpack Settings` (title case).

To test:
- Launch the app and go to `My Site` tab.
- Make sure that the above mentioned `string` changes are as described.
- Go to `My Site` -> `Toolbar Avatar` -> `App Settings` -> `Test feature configuration`.
- Enable `MySiteImprovementsFeatureConfig`, scroll down and click the `RESTART THE APP` button.
- Make sure that the above mentioned `string` changes are as described.

**Merge Instructions**
1. Make sure this PR has been reviewed by design.
1. Remove the `Needs Design Review` label.
1. Merge this PR.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by zwarm — APPROVED** [review]
LGTM 🙇

**Comment by ParaskP7** [comment]
👋 @osullivanchris !

Let us know if you are okay with this change and we will then merge that to `develop` to get it be part of the next release (`16.7`).

👋 @zwarm !

Since I believe Chris will be probably okay with this change, as he requested those on #13969, please merge this PR by the end of your day (if it is still pending review by Chris), so that in ends up in the next release (`16.7`). I am pretty confident (99.9%) that Chris will be okay with it anyway. In any case we can make another change if need be and add it to the Beta release next sprint.

**Review by osullivanchris — APPROVED** [review]
thanks for the fix @ParaskP7 !

**Comment by ashiagr** [comment]
Going ahead and merging this one to target 16.7!

**Pull request merged**
